log/log.go

```diff
@@ -1,71 +1,118 @@
-	std.hooks = append(std.hooks, hook)
-	std.base.Hooks.Add(hook)
+	std = std.WithOptions(zap.Hooks(hook.OnEntry))
 }
 
 // Logging methods
 
-func Debug(args ...interface{})                             { std.Debug(args...) }
-func Debugf(format string, args ...interface{})             { std.Debugf(format, args...) }
-func Debugln(args ...interface{})                           { std.Debugln(args...) }
-func Error(args ...interface{})                             { std.Error(args...) }
-func Errorf(format string, args ...interface{})             { std.Errorf(format, args...) }
-func Errorln(args ...interface{})                           { std.Errorln(args...) }
-func Info(args ...interface{})                              { std.Info(args...) }
-func Infof(format string, args ...interface{})              { std.Infof(format, args...) }
-func Infoln(args ...interface{})                            { std.Infoln(args...) }
-func Print(args ...interface{})                             { std.Print(args...) }
-func Printf(format string, args ...interface{})             { std.Printf(format, args...) }
-func Println(args ...interface{})                           { std.Println(args...) }
-func Fatal(args ...interface{})                             { std.Fatal(args...) }
-func Fatalf(format string, args ...interface{})             { std.Fatalf(format, args...) }
-func Fatalln(args ...interface{})                           { std.Fatalln(args...) }
-func Warn(args ...interface{})                              { std.Warn(args...) }
-func Warnf(format string, args ...interface{})              { std.Warnf(format, args...) }
-func Warnln(args ...interface{})                            { std.Warnln(args...) }
-func WithField(key string, value interface{}) *logrus.Entry { return std.WithField(key, value) }
-func WithFields(fields logrus.Fields) *logrus.Entry         { return std.WithFields(fields) }
+func Print(args ...interface{})                 { Info(args...) }
+func Printf(format string, args ...interface{}) { Infof(format, args...) }
+func Println(args ...interface{})               { Infoln(args...) }
+
+func Debug(args ...interface{}) { std.Debug(fmt.Sprint(args...)) }
+func Error(args ...interface{}) { std.Error(fmt.Sprint(args...)) }
+func Info(args ...interface{})  { std.Info(fmt.Sprint(args...)) }
+func Warn(args ...interface{})  { std.Warn(fmt.Sprint(args...)) }
+func Fatal(args ...interface{}) { std.Fatal(fmt.Sprint(args...)) }
+
+func Debugf(format string, args ...interface{}) { std.Debug(fmt.Sprintf(format, args...)) }
+func Errorf(format string, args ...interface{}) { std.Error(fmt.Sprintf(format, args...)) }
+func Infof(format string, args ...interface{})  { std.Info(fmt.Sprintf(format, args...)) }
+func Warnf(format string, args ...interface{})  { std.Warn(fmt.Sprintf(format, args...)) }
+func Fatalf(format string, args ...interface{}) { std.Fatal(fmt.Sprintf(format, args...)) }
+
+func Debugln(args ...interface{}) { std.Debug(fmt.Sprintln(args...)) }
```

Comment:
Should `*ln` variant be really available here? IMO, better don't make consumer think about newlines.

Reply:
here's the difference: https://play.golang.org/p/2nWgpe-aOn

Reply:
Doesn't look like killer feature, tbh, but fine